src/recursion.js

```diff
@@ -3,16 +3,52 @@
 const nFibonacci = (n) => {
   // fibonacci sequence: 1 2 3 5 8 13 ...
   // return the nth number in the sequence
+  if (n <= 1) return 1;
+  return nFibonacci(n - 1) + nFibonacci(n - 2);
 };
 
 const nFactorial = (n) => {
   // factorial example: !5 = 5 * 4 * 3 * 2 * 1
   // return the factorial of `n`
+  if (n < 1) return 1;
+  return n * nFactorial(n - 1);
 };
 
 const checkMatchingLeaves = (obj) => {
   // return true if every property on `obj` is the same
   // otherwise return false
+  //                                             { key: value, key:  vv value vv }
+  // INPUT: will either be a number or an object {   x:   1  ,   y: {key: value} }
```

Comment:
Nice commenting going on here to lay out what your algorithm needs to do. Definitely a good habit to get into.